app/workers/database.php

```diff
@@ -120,6 +120,28 @@ protected function deleteAttribute(Document $collection, Document $attribute, st
             $dbForInternal->updateDocument('attributes', $attribute->getId(), $attribute->setAttribute('status', 'failed'));
         }
 
+        // the underlying database removes/rebuilds indexes when attribute is removed
+        // update indexes table with changes
+        /** @var Document[] $indexes */
+        $indexes = $collection->getAttribute('indexes', []);
+
+        foreach ($indexes as $index) {
```

Comment:
Why are we updating or deleting the index, shouldn't we always delete any relevant index when at least on attribute is missing?

Reply:
I'm pretty sure that at least for mariadb the index never gets rebuild this way, but completely removed.🤔

Reply:
According to my research and testing, MariaDB doesn't drop the whole index. From the docs:
> If the column is part of any index, the column will be dropped from them, except if you add a new column with identical name at the same time. The index will be dropped if all columns from the index were dropped.
[ref](https://mariadb.com/kb/en/alter-table/#drop-column)

I tested this manually with the following schema:
![mariadbdropindex](https://user-images.githubusercontent.com/9708641/133933648-0363a47b-cd9f-4524-b7dd-9742f8cbb501.png)

I then dropped column `one`, and the indexes were rebuilt accordingly:
![droppedonecolumn](https://user-images.githubusercontent.com/9708641/133933664-b40e5852-7fc0-4718-9fa0-20668da82447.png)

Reply:
Going to add logic for the edge case where `three` is deleted and we end up with duplicate indexes:
![indexedgecase](https://user-images.githubusercontent.com/9708641/134047172-1bc07e13-e1e2-4a2a-9e6d-ca752a96163b.png)

Reply:
☝🏻 edge case handled in c960116 and tested in fdd4383